lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2743,23 +2785,27 @@ static SILValue createHostSend(SILBuilder &B, SILLocation loc, SILValue value,
 
   auto &ctx = B.getASTContext();
   Type scalarValueTy, tensorValueTy;
-  if (!isTensorHandle(value->getType().getASTType())) {
+  if (isTensorHandle(value->getType().getASTType())) {
+    tensorValueTy = value->getType().getASTType();
+    scalarValueTy = getTensorHandleElementType(tensorValueTy);
+  } else {
     assert(createScalarTensorFn);
     // Here scalar type is something like $Builtin.FPIEEE32 -- convert it to an
     // AccelerableByTensorFlow conforming type like Float first, and then create
     // a scalar tensor to send that value.
     scalarValueTy = value->getType().getASTType();
     if (!scalarValueTy->getAs<StructType>()) {
-      // The value must be defined by a struct_extract like:
-      //   %34 = struct_extract %33 : $Float, #Float._value
-      //
-      // In this case we set `value` to the Float operand like %33 above.
-      auto *SEI =
-          castWithDebugInfo<StructExtractInst>(value->getDefiningInstruction());
-      assert(SEI->getFieldNo() == 0);
-      value = SEI->getOperand();
+      assert(value->getDefiningInstruction());
+      auto *typeDecl =
+          getStdlibNumericTypeDeclFromBuiltinType(scalarValueTy, ctx);
+      if (!typeDecl) {
+        value->dump();
```

Comment:
Will it dump in release mode?

Reply:
I intend to keep it even in release mode, because otherwise we could get undefined behavior. This is analogous to `internalConsistencyCheck()` that we had in S4TF stdlib and runtime code. Once our infra becomes mature enough, we can revisit this.